site/pages/api/publish.api.ts

```diff
@@ -0,0 +1,179 @@
+                : {}),
+            Consequences: {
+                Consequence: [
+                    {
+                        Condition: "unknown",
+                        Severity: consequenceData.disruptionSeverity,
+                        Affects: {
+                            ...(consequenceData.consequenceType === "networkWide" ||
+                            consequenceData.consequenceType === "operatorWide"
+                                ? {
+                                      Networks: {
+                                          AffectedNetwork: {
+                                              VehicleMode: consequenceData.vehicleMode,
+                                              AllLines: "",
+                                          },
+                                      },
+                                  }
+                                : {}),
+                            ...(consequenceData.consequenceType === "operatorWide"
+                                ? {
+                                      Operators: {
+                                          AffectedOperator: {
+                                              OperatorRef: consequenceData.consequenceOperator,
+                                          },
+                                      },
+                                  }
+                                : {}),
+                        },
+                        Advice: {
+                            Details: consequenceData.description,
+                        },
+                        Blocking: {
+                            JourneyPlanner: consequenceData.removeFromJourneyPlanners === "yes",
+                        },
+                        ...(consequenceData.disruptionDelay
+                            ? { Delays: { Delay: `PT${consequenceData.disruptionDelay}M` } }
+                            : {}),
+                    },
+                ],
+            },
+        };
+
+        let completePtSituationElement: PtSituationElement;
+
+        if (isMiscellaneousReason(reason)) {
+            completePtSituationElement = {
+                ...ptSituationElement,
+                ReasonType: "MiscellaneousReason",
+                MiscellaneousReason: reason,
+            };
+        } else if (isEnvironmentReason(reason)) {
+            completePtSituationElement = {
+                ...ptSituationElement,
+                ReasonType: "EnvironmentReason",
+                EnvironmentReason: reason,
+            };
+        } else if (isPersonnelReason(reason)) {
+            completePtSituationElement = {
+                ...ptSituationElement,
+                ReasonType: "PersonnelReason",
+                PersonnelReason: reason,
+            };
+        } else {
+            completePtSituationElement = {
+                ...ptSituationElement,
+                ReasonType: "EquipmentReason",
+                EquipmentReason: reason,
+            };
+        }
```

Comment:
Not that this is a big deal, but I wonder if a switch statement is more appropriate for this?